lucene/core/src/java/org/apache/lucene/index/IndexWriter.java

```diff
@@ -3234,15 +3234,23 @@ private long prepareCommitInternal() throws IOException {
               // corresponding add from an updateDocument) can
               // sneak into the commit point:
               toCommit = segmentInfos.clone();
-
+              pendingCommitChangeCount = changeCount.get();
```

Comment:
I moved these above the updatePendingMerges. not sure it's a problem but looks better from a readers perspective to incRef before we do anything to it potentially.